pkg/strategy/grid2/strategy.go

```diff
@@ -1413,10 +1462,10 @@ func (s *Strategy) recoverGridWithOpenOrdersByScanningTrades(ctx context.Context
 // buildPinOrderMap build the pin-order map with grid and open orders.
 // The keys of this map contains all required pins of this grid.
 // If the Order of the pin is empty types.Order (OrderID == 0), it means there is no open orders at this pin.
-func (s *Strategy) buildPinOrderMap(grid *Grid, openOrders []types.Order) (map[string]types.Order, error) {
-	pinOrderMap := make(map[string]types.Order)
+func (s *Strategy) buildPinOrderMap(pins []Pin, openOrders []types.Order) (PinOrderMap, error) {
+	pinOrderMap := make(PinOrderMap)
 
-	for _, pin := range grid.Pins {
+	for _, pin := range pins {
 		priceStr := s.FormatPrice(fixedpoint.Value(pin))
```

Comment:
can you move FormatPrice func to types.Market? so that you don't need to depend on the strategy object.